scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import time
+import urllib.parse
+from argparse import ArgumentParser, Namespace
+from collections import namedtuple
+from concurrent.futures import ProcessPoolExecutor
+from typing import Iterable, List, Optional, Tuple, Union
+from uuid import uuid4
+
+# PB2 stuff
+import google.protobuf.any_pb2 as any_pb2
+import lz4.frame
+import pandas as pd
+import pyarrow as pa
+import pyspark.sql.connect.proto as pb2
+import pyspark.sql.connect.proto.cloud_pb2 as cloud_pb2
+import requests
+from databricks import sql
+from databricks.connect import DatabricksSession
+from databricks.sdk import WorkspaceClient
+from databricks.sql.client import Connection as Connection
+from databricks.sql.client import Cursor as Cursor
+from packaging import version
+from pyspark.sql import SparkSession
+from pyspark.sql.connect.client.core import SparkConnectClient
+from pyspark.sql.connect.client.reattach import \
+    ExecutePlanResponseReattachableIterator
+from pyspark.sql.connect.dataframe import DataFrame
+from pyspark.sql.dataframe import DataFrame as SparkDataFrame
+from pyspark.sql.types import Row
+
+MINIMUM_DBR_VERSION = '14.0.0'
+
+log = logging.getLogger(__name__)
+
+Result = namedtuple(
+    'Result', ['url', 'row_count', 'compressed_size', 'uncompressed_size'
+              ])  # pyright: ignore
+
+cf_collect_type = 'arrow'  # optionally change to json if arrow fails
+
+# This is a monkey patch on top of the DB Connect package that allows
```

Comment:
could you specify a little bit more how the monkeypatch differs from the code it is replacing? Is it a complete rewrite or just a few lines? If just a few lines, please demarcate them inside the function with comments.

Reply:
It is a new feature, add-on to the existing package.